### PR TITLE
Eclipse timeout

### LINKED ISSUE #191: Eclipse download timeout
Several people reported Eclipse bundle fetch timed out. Let's see if we can make this better.

### PR DESCRIPTION
**Pull request by Advill**
closes #191 

Have script set up to timeout after 30 seconds, as well as a backup to try to pull from the vt mirror.

### CODE REVIEW

File: roles/eclipse/vars/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ eclipse:
   # Ensure the URL does not specify a mirror and that &r=1 is at the end, which
   # directly links to the file and not the web page with a download button
   url: 'https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/photon/R/eclipse-java-photon-R-linux-gtk-x86_64.tar.gz&r=1'
+  urlbck: 'https://www.eclipse.org/downloads/download.php?file=/oomph/epp/photon/R/eclipse-inst-linux64.tar.gz&mirror_id=504'
```
Comment (kylelaker) [inline roles/eclipse/vars/main.yml]: We need the URL for Eclipse for Java Developers, not the installer, so it should have a URL with `eclipse-java-photon-R-linux-gtk-x86_64.tar.gz` somewhere in it.

Additionally, can this variable use a longer name without an abbreviation like `backup_url` or `url_backup`?

File: roles/eclipse/tasks/main.yml [hunk 2]
```diff
@@ -11,7 +11,17 @@
       get_url:
         url: '{{ eclipse.url }}'
         dest: '{{ eclipse.zip }}'
+        timeout: 30
         force: yes
+      register: eclout
+      ignore_errors: true
+    - name: Fetch Eclipse bundle from backup server
+      get_url:
+        url: '{{ eclipse.urlbck }}'
```
Comment (kylelaker) [inline roles/eclipse/tasks/main.yml]: See comment on naming this variable

File: roles/eclipse/vars/main.yml (hunk 1 above)
Comment (Advill) [reply roles/eclipse/vars/main.yml]: ah yes, I'll get that set up

File: roles/eclipse/tasks/main.yml (hunk 2 above)
Comment (Advill) [reply roles/eclipse/tasks/main.yml]: Will do

**Comment by kylelaker** [comment]
@Advill I believe the backup URL we'd want to use is:

http://mirror.cc.vt.edu/pub/eclipse/technology/epp/downloads/release/photon/R/eclipse-java-photon-R-linux-gtk-x86_64.tar.gz

**Comment by ripleymj** [comment]
I keep seeing references to "2018-09 R", instead of Photon. Can you see if we need to upgrade this while you're in there tinkering?

**Comment by kylelaker** [comment]
@ripleymj It seems the Eclipse has moved to a new release cycle. We need to determine which release we want to include. That may be beyond the scope of this PR.

https://wiki.eclipse.org/Simultaneous_Release

File: roles/eclipse/vars/main.yml (hunk 1 above)
Comment (ripleymj) [reply roles/eclipse/vars/main.yml]: In the looping rewrite, I was encouraging prefixing variable names with the role name, as they're all global.

**Pull request merged**